commands/benchmark.go

```diff
@@ -0,0 +1,330 @@
+package commands
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"github.com/chris-wood/dns"
+	"github.com/chris-wood/odoh"
+	"github.com/urfave/cli"
+	"log"
+	mathrand "math/rand"
+	"net/http"
+	"os"
+	"time"
+)
+
+// This RunningTime structure contains the epoch timestamps for each of the operations
+// taking place. The explanations are as follows:
+// 1. Start => Epoch time at which the client starts to prepare the question
+// 2. ClientHashingOverheadTime => Epoch time at which the client hashes the symmetric key used for request identification.
+// 3. ClientQueryEncryptionTime => Epoch time at which the client completes the encryption and serialization of the question.
+// 4. ClientUpstreamRequestTime => Epoch time indicating the start of the network request.
+// 5. ClientDownstreamResponseTime => Epoch time indicating the receipt of the response and deserialization into ObliviousDNSMessage
+// 6. EndTime => Epoch time indicating the end of all tasks for the request.
+// NOTE: All timestamps are stored in NanoSecond granularity and need to be converted into microseconds (/1000.0) or milliseconds (/1000.0^2)
+type RunningTime struct {
```

Comment:
Yup, since we marshal these to JSON, they don't work correctly if the type and the members of the struct aren't exported. Example:
```
m := message{
    id:   11,
    name: "apple.com",
}

x, _ := json.Marshal(m)
fmt.Printf("%v\n", x)
```